openff/toolkit/topology/topology.py

```diff
@@ -2385,3 +2386,25 @@ def hierarchy_iterator(
             if hasattr(molecule, iter_name):
                 for item in getattr(molecule, iter_name):
                     yield item
+
+    def __getattr__(self, name: str) -> List["HierarchyElement"]:
+        """If a requested attribute is not found, check the hierarchy schemes"""
+        # Avoid attempting to process dunder methods as hierarchy scheme iterator names
+        if name.startswith("__"):
+            raise AttributeError
+
```

Comment:
This is not my proudest work but I haven't found an issue with it yet and some other workarounds I tinkered with were more complex and less functional. This flared up in tests when `Topology.__add__` or something similar called `deepcopy()` on a topology, which for one reason or another checks to see if `Topology.__deepcopy__` exists (it does not). AFAICT the implementation here is the same with `name.startswith("_")`.

The best argument one could make for this is pointing out that `__getattr__` is only called as a last resort when an attribute it not found on an object, and we don't want any dunder methods to be processed as hierarchy scheme iterator names anyway.